pkg/environment/env.go

```diff
@@ -215,7 +215,17 @@ func (env *Environment) PlatformString() string {
 // SelfUpdate updates TiUp.
 func (env *Environment) SelfUpdate() error {
 	if env.v1Repo != nil {
-		return env.v1Repo.DownloadTiup(env.LocalPath("bin"))
+		if err := env.v1Repo.DownloadTiup(env.LocalPath("bin")); err != nil {
+			return err
+		}
+
+		// Cover the root.json from tiup.bar.gz
+		profile := localdata.InitProfile()
+		if err := profile.ResetMirror(Mirror(), ""); err != nil {
+			return err
+		}
+
+		return nil
```

Comment:
The +224 to +228 can be simplified to `return profile.ResetMirror(Mirror(), "")`